flink-runtime/src/main/java/org/apache/flink/runtime/jobmanager/DefaultExecutionPlanStore.java

```diff
@@ -109,153 +110,163 @@ public void stop() throws Exception {
+                        LOG.warn(
+                                "{} already exists in {}.",
+                                executionPlan,
+                                executionPlanStateHandleStore);
                     }
-                } else if (addedJobGraphs.contains(jobID)) {
+                } else if (addedExecutionPlans.contains(jobID)) {
                     try {
-                        jobGraphStateHandleStore.replace(name, currentVersion, jobGraph);
-                        LOG.info("Updated {} in {}.", jobGraph, getClass().getSimpleName());
+                        executionPlanStateHandleStore.replace(name, currentVersion, executionPlan);
+                        LOG.info("Updated {} in {}.", executionPlan, getClass().getSimpleName());
 
                         success = true;
                     } catch (StateHandleStore.NotExistException ignored) {
-                        LOG.warn("{} does not exists in {}.", jobGraph, jobGraphStateHandleStore);
+                        LOG.warn(
+                                "{} does not exists in {}.",
+                                executionPlan,
+                                executionPlanStateHandleStore);
                     }
                 } else {
                     throw new IllegalStateException(
-                            "Trying to update a graph you didn't "
-                                    + "#getAllSubmittedJobGraphs() or #putJobGraph() yourself before.");
+                            "Trying to update a execution plan you didn't "
+                                    + "#getAllSubmittedExecutionPlans() or #putExecutionPlan() yourself before.");
                 }
             }
         }
 
-        LOG.info("Added {} to {}.", jobGraph, jobGraphStateHandleStore);
+        LOG.info("Added {} to {}.", executionPlan, executionPlanStateHandleStore);
     }
 
     @Override
     public void putJobResourceRequirements(
             JobID jobId, JobResourceRequirements jobResourceRequirements) throws Exception {
         synchronized (lock) {
-            @Nullable final JobGraph jobGraph = recoverJobGraph(jobId);
-            if (jobGraph == null) {
+            @Nullable final ExecutionPlan executionPlan = recoverExecutionPlan(jobId);
+            if (executionPlan == null) {
                 throw new NoSuchElementException(
                         String.format(
-                                "JobGraph for job [%s] was not found in JobGraphStore and is needed for attaching JobResourceRequirements.",
+                                "ExecutionPlan for job [%s] was not found in ExecutionPlanStore and is needed for attaching JobResourceRequirements.",
                                 jobId));
             }
-            JobResourceRequirements.writeToJobGraph(jobGraph, jobResourceRequirements);
-            putJobGraph(jobGraph);
+
+            checkState(executionPlan instanceof JobGraph);
```

Comment:
Is this method expected to be only invoked on a `JobGraph`?

Reply:
Yes, the JobResourceRequirements class is designed specifically for JobGraph

Reply:
Seems exceptions will happen if the `StreamGraph` of a streaming job is stored and `updateJobResourceRequirements(...)` is invoked.

Also even if it is a `StreamGraph`, this configuration setting can be applied via `jobConfiguration`?

Reply:
Yes, you're right. I have corrected it.